### ISSUE TITLE
Failures in Docker Container

### ISSUE THREAD

**Issue by tatoalo** [Fluxticks/TikTokDL#6]
Hey there 👋🏻 ,

I was looking into dockerize your project for some testing and saw this kind of failures:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/tiktokdl/download_post.py", line 156, in __get_post
    request_value = await request.value
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_async_base.py", line 34, in value
    return mapping.from_maybe_impl(await self._future)
playwright._impl._errors.TimeoutError: Timeout 20000ms exceeded while waiting for event "request"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
    post_info = asyncio.run(
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/dist-packages/tiktokdl/download_post.py", line 228, in get_post
    raise e
  File "/usr/local/lib/python3.10/dist-packages/tiktokdl/download_post.py", line 212, in get_post
    result = await __get_post(url=url,
  File "/usr/local/lib/python3.10/dist-packages/tiktokdl/download_post.py", line 135, in __get_post
    async with async_playwright() as playwright:
  File "/usr/local/lib/python3.10/dist-packages/playwright/async_api/_context_manager.py", line 58, in __aexit__
    await self._connection.stop_async()
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_connection.py", line 289, in stop_async
    self.cleanup()
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_connection.py", line 300, in cleanup
    callback.future.set_exception(self._closed_error)
asyncio.exceptions.InvalidStateError: invalid state
```

For additional context, I've used an `ubuntu:22.04` image.

I know perhaps it's out of your scope for the project, but do you know how would I be able to run this in a docker container? Thanks a lot!

**Comment by Fluxticks** [comment]
Hiya!

Thanks for letting me know about the error!

The issue you've mentioned is a more general Playwright error when the expected network request does not occur. This should have been caught by the retry logic, but I had implemented it slightly naively. I've updated the logic to try to properly retry when this error occurs (amongst others), and have released [v1.6.0](https://github.com/Fluxticks/TikTokDL/releases/tag/v1.6.0) that hopefully addresses the issue you are experiencing :)

**Comment by tatoalo** [comment]
Thanks a lot @Fluxticks for looking into this so quickly 🙏🏻 

I'm now experiencing the following:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/tiktokdl/download_post.py", line 215, in get_post
    result = await __get_post(url=url,
  File "/usr/local/lib/python3.10/dist-packages/tiktokdl/download_post.py", line 161, in __get_post
    data = await response.json()
  File "/usr/local/lib/python3.10/dist-packages/playwright/async_api/_generated.py", line 636, in json
    return mapping.from_maybe_impl(await self._impl_obj.json())
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_network.py", line 616, in json
    return json.loads(await self.text())
  File "/usr/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
    post_info = asyncio.run(
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/dist-packages/tiktokdl/download_post.py", line 229, in get_post
    raise RetryLimitReached(e, retries, url)
tiktokdl.exceptions.RetryLimitReached: The maximum number of retries was reached (10). The last error that occured was:

 JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

It seems like that if the request is being made from a docker container, there's something wrong at the network level? This is pretty weird 🤔

**Comment by Fluxticks** [comment]
After some digging; The response from the server is giving a content-length of 0. It also seems to be giving a header that, if I'm understanding correctly, is related to CAPTCHA verification. I have since found that TikTok have updated what they want regarding CAPTCHA verification so could be related. Once I've fixed that and confirmed that it is reliably solving the CAPTCHA again, I'll see if the response gives the expected data. 

Im still not sure why this is cropping up only in Docker environments, but hopefully it gets solved with the CAPTCHA updates
